### DISCUSSION TITLE
When should I use the `json` function on `fetch` responses?

### DISCUSSION THREAD

**Discussion by fedemp** [remix-run/remix#5620] [Q&A] [answered]
In the [External APIS](https://remix.run/docs/en/1.13.0/guides/data-loading#external-apis) section of the docs, I noticed this example.

```
export async function loader() {
  const res = await fetch("https://api.github.com/gists");
  return json(await res.json());
}
```

What's the point of wrapping the `res.json()` promise in the `json` method`? Can't I just return the resolved JSON promise? i.e.

```
const data = await res.json();
return data;
```

Cause as I understand it, any information from the original `fetch` response is lost in the first example too anyway.

**Comment by lili21** [comment]
Event if you don't use `json`, Remix would do it for you.

I would recommend to use it all the time. it would be more easier if you want to set headers or other Response options.

```js
return json(data);

// add headers
return json(data, { headers: { xxx }} )
```

**Comment by sergiodxa** [comment, marked as answer]
The difference is that `await res.json()` returns a JS object with the data, while `json` helper returns a Response object.

As @lili21 mentioned, if you don't return a Response Remix will pass your data to the `json` helper and create it, I think it's a good practice to always use `json` since you can be explicit that you're returning a json response and you can later add headers if you want without many changes.

Additionally, you could do `return await fetch("https://api.github.com/gists")` too, because the resolved value of a fetch is a Response, but if you do this you will also send any header that Response has, for example Cache-Control headers, this is something you typically don't want unless you're doing a proxy, so you better call `res.json()` first.
